src/generators/java/JavaConstrainer.ts

```diff
@@ -84,6 +84,10 @@ export const JavaDefaultTypeMapping: TypeMapping<JavaOptions> = {
     return 'Object';
   },
   Dictionary ({constrainedModel}): string {
+    //Limitations to Java is that maps cannot have specific value types...
+    if (constrainedModel.value.type === 'int') {
+      constrainedModel.value.type = 'Integer';
```

Comment:
Here, you override `constrainedModel` value type. Should we create local variable for such a thing?

Reply:
I would have liked that it was originally constrained in the `ConstrainedIntegerModel` callback, but it does not have access to parent models (maybe it should 🤔) which means this is the only way to do it currently.

It needs to overwrite the type because if any preset access the value model type for the dictionary, as part of some function, it needs to return `Integer`, always.

Reply:
Thanks for clarification!